sdk/translation/ai-translation-text-rest/src/customClient.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { getClient, ClientOptions } from "@azure-rest/core-client";
+import * as coreRestPipeline from "@azure/core-rest-pipeline";
+import { PipelineRequest, PipelineResponse, SendRequest } from "@azure/core-rest-pipeline";
+import { TextTranslationClient } from "./generated/clientDefinitions";
+import {
+  TranslatorCredential,
+  TranslatorAuthenticationPolicy,
+  TranslatorAzureKeyAuthenticationPolicy,
+} from "./authentication";
+import { AzureKeyCredential, TokenCredential } from "@azure/core-auth";
+
+const DEFAULT_SCOPE = "https://cognitiveservices.azure.com/.default";
+const DEFAULT_ENPOINT = "https://api.cognitive.microsofttranslator.com";
+const PLATFORM_HOST = "cognitiveservices";
+const PLATFORM_PATH = "/translator/text/v3.0";
+
+/** Policy that sets the api-version (or equivalent) to reflect the library version. */
+const apiVersionPolicy = {
+  name: "MTApiVersionPolicy",
+  async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
+    const param = request.url.split("?");
+    if (param.length > 1) {
+      const newParams = param[1].split("&");
+      newParams.push("api-version=3.0");
+      request.url = param[0] + "?" + newParams.join("&");
+    } else {
+      // no query parameters in request url
+      request.url = param[0] + "?api-version=3.0";
+    }
+    return next(request);
+  },
+};
+
+/**
+ * Initialize a new instance of `TextTranslationClient`
+ * @param endpoint type: string, Supported Text Translation endpoints (protocol and hostname, for example:
+ *     https://api.cognitive.microsofttranslator.com).
+ * @param options type: ClientOptions, the parameter for all optional parameters
+ */
+export default function createClient(
+  endpoint: undefined | string,
+  credential: undefined | TranslatorCredential | AzureKeyCredential | TokenCredential = undefined,
+  options: ClientOptions = {}
+): TextTranslationClient {
```

Comment:
We should create overloads for create client. Otherwise if I users want to use the default endpoint they will be forced to pass undefined as the first positional parameter. 

We'll need these overloads

```ts
export default function createClient(
  endpoint: string,
  credential: TranslatorCredential | AzureKeyCredential | TokenCredential,
  options?: ClientOptions
): TextTranslationClient;
export default function createClient(
  endpoint: string,
  options?: ClientOptions
): TextTranslationClient;
export default function createClient(
  credential: TranslatorCredential | AzureKeyCredential | TokenCredential,
  options?: ClientOptions
): TextTranslationClient;
```

Reply:
I will add it to the list of items we need to fix before GA.

Reply:
https://github.com/Azure/azure-sdk-for-js/issues/25545